### PR TITLE
prov/efa: Make EFA unit tests build with make, but not be installed

### PR DESCRIPTION
**Pull request by a-szegel**
Signed-off-by: Seth Zegelstein <szegel@amazon.com>

### CODE REVIEW

File: prov/efa/Makefile.include [hunk 1]
```diff
@@ -82,6 +82,7 @@ _efa_headers = \
 
 if ENABLE_EFA_UNIT_TEST
 check_PROGRAMS += prov/efa/test/efa_unit_test
```
Comment (wzamazon) [inline prov/efa/Makefile.include:84]: Do we still need this the `efa_unit_test` in `check_PROGRAMS`?

Comment (a-szegel) [reply prov/efa/Makefile.include]: done

**Review by wzamazon — APPROVED** [review]

**Comment by wzamazon** [comment]
@shefty Can you help getting error message of the "continuous-integration/jenkins/pr-head" CI error?

**Comment by shefty** [comment]
The CI aborted for some reason not captured in the logs.  Maybe the PR was updated while CI was running, causing some issue canceling the old run while the new one started?  I restarted.

**Pull request merged**